intera_interface/src/intera_interface/camera.py

```diff
@@ -26,28 +33,65 @@ def __init__(self):
         """
         Constructor.
         """
+        self._node_config = None
+        self._node_config_sub = rospy.Subscriber('io/internal_camera/config', IONodeConfiguration, self._node_config_cb)
+        self.cameras_io = dict()
+
         camera_param_dict = RobotParams().get_camera_details()
         camera_list = camera_param_dict.keys()
         # check to make sure cameras is not an empty list
         if not camera_list:
             rospy.logerr(' '.join(["camera list is empty: ", ' , '.join(camera_list)]))
             return
 
+        intera_dataflow.wait_for(
+            lambda: self._node_config is not None,
+            timeout=5.0,
+            timeout_msg=("Failed to connect to Camera Node and retrieve configuration.")
+        )
+        cameras_to_load = self._get_camera_launch_config().keys()
+
         camera_capabilities = {
             "mono": ['cognex'],
             "color": ['ienso_ethernet'],
             "auto_exposure": ['ienso_ethernet'],
             "auto_gain": ['ienso_ethernet']
         }
-        self.cameras_io = dict()
         for camera in camera_list:
             cameraType = camera_param_dict[camera]['cameraType']
-            self.cameras_io[camera] = {
-                'interface': IODeviceInterface("internal_camera", camera),
-                'is_color': (cameraType in camera_capabilities['color']),
-                'has_auto_exposure': (cameraType in camera_capabilities['auto_exposure']),
-                'has_auto_gain': (cameraType in camera_capabilities['auto_gain']),
-            }
+            try:
+                interface = IODeviceInterface("internal_camera", camera)
+
+                self.cameras_io[camera] = {
+                    'interface': interface,
+                    'is_color': (cameraType in camera_capabilities['color']),
+                    'has_auto_exposure': (cameraType in camera_capabilities['auto_exposure']),
+                    'has_auto_gain': (cameraType in camera_capabilities['auto_gain']),
+                }
+            except OSError as e:
+                if camera not in cameras_to_load:
+                    rospy.logwarn("Expected camera ({0}) is not configured to launch in this run"
+                    " configuration. Make sure you are running in SDK Mode.".format(camera))
+                else:
+                    rospy.logfatal("Could not find expected camera ({0}) for this robot.\n"
```

Comment:
I suppose this isn't fatal, but an error. We allow the camera enumeration to continue despite the missing camera.

Reply:
Yeah, I was considering what level it should be. It does point to a likely hardware-level, completely unusable state ... for that camera. But it won't prevent the whole robot from not working, and may not prevent the example from working, depending on which camera you want to use.

Reply:
You know what, I will change it to 'ERROR' level. It's still red and bad, and I honestly don't think anyone will ever notice the difference. Besides it doesn't immediately kill the program execution, so that's probably a pretty good clue too.